mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -0,0 +1,67 @@
+package acctest
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+)
+
+// Hardcode the Terraform resource name -> API service name mapping temporarily.
+// TODO: [tgc] read the mapping from the resource metadata files.
+var ApiServiceNames = map[string]string{
+	"google_compute_instance": "compute.googleapis.com",
+	"google_project":          "cloudresourcemanager.googleapis.com",
+}
+
+// Gets the test metadata for tgc:
+//   - test config
+//   - cai asset name
+//     For example: //compute.googleapis.com/projects/ci-test-188019/zones/us-central1-a/instances/tf-test-mi3fqaucf8
+func GetTestMetadataForTgc(service, resourceType, resourceName, config string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		address := fmt.Sprintf("%s.%s", resourceType, resourceName)
```

Comment:
what would you think about passing in the address instead, and then breaking it into the resource type and resource name? That might make the signature more similar to other checks (which usually take an address).

(I have a slight inclination to pass the full address but don't care strongly about it.)

Reply:
Passing the address works here to be consistent with other checks.